tests/test_h5md.py

```diff
@@ -0,0 +1,190 @@
+#
+# Copyright The NOMAD Authors.
+#
+# This file is part of NOMAD. See https://nomad-lab.eu for further info.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import pytest
+import numpy as np
+
+from nomad.datamodel import EntryArchive
+from atomisticparsers.h5md import H5MDParser
+
+
+def approx(value, abs=0, rel=1e-6):
+    return pytest.approx(value, abs=abs, rel=rel)
+
+
+@pytest.fixture(scope='module')
+def parser():
+    return H5MDParser()
+
+
+def test_md(parser):
+    archive = EntryArchive()
+    parser.parse('tests/data/h5md/openmm/test_traj_openmm_5frames.h5', archive, None)
+
+    sec_run = archive.run[0]
+    assert sec_run.program.name == 'OpenMM'
+    assert sec_run.program.version == '-1.-1.-1'
+    assert len(sec_run.x_h5md_version) == 2
+    assert sec_run.x_h5md_version[1] == 0
+    assert sec_run.x_h5md_author.name == 'Joseph F. Rudzinski'
+    assert sec_run.x_h5md_author.email == 'joseph.rudzinski@physik.hu-berlin.de'
+    assert sec_run.x_h5md_creator.name == 'h5py'
+    assert sec_run.x_h5md_creator.version == '3.6.0'
+
+    sec_method = sec_run.method
+    sec_atom_params = sec_method[0].atom_parameters
+    assert len(sec_atom_params) == 31583
+    assert sec_atom_params[164].label == 'O'
+    assert sec_atom_params[164].mass.magnitude == approx(2.656767806475139e-26)
```

Comment:
sorry if I will make this more difficult for you but we should compare the number that is parsed , so you would have mass.to(h5unit).magnitude

Reply:
actually, there is no inherent h5md unit. The user can specify the unit using the UnitRegistry-compatible strings.

I'm curious though why we need to compare in the "native" units when applicable (this is certainly not what the gromacs and lammps tests do currently)? If there was a mistake in the parsing we will certainly notice after conversion to a different unit, right?

Reply:
I think it is simply more convenient if we compare to the value that is 'printed on the mainfile'. Yes, this is not what is currently done in most parsers but we are trying to move in this direction. This is a remnant of the old infra where we do not support unit conversions.

Reply:
Do you mean here simply to convert the value within the test to the unit found in the original file and then enter the value from the file (instead of from the parsed result)?

Reply:
yes, do something like, quantity.to('orignal_unit').magnitude == approx(raw_value_in_file)

Reply:
done.